src/facade/dragonfly_connection.cc

```diff
@@ -337,8 +337,17 @@ void Connection::HandleRequests() {
   ThisFiber::SetName("DflyConnection");
 
   if (absl::GetFlag(FLAGS_tcp_nodelay)) {
-    int val = 1;
-    CHECK_EQ(0, setsockopt(socket_->native_handle(), IPPROTO_TCP, TCP_NODELAY, &val, sizeof(val)));
+    int val = 0;
+
+    // Get the underlying address family first to avoid setting tcp options on unix domain sockets
+    socklen_t bufsize = sizeof(val);
+    int res = getsockopt(socket_->native_handle(), SOL_SOCKET, SO_DOMAIN, &val, &bufsize);
+    LOG_IF(ERROR, res != 0) << "Bad getsockopt call " << res;
+
+    if (res == 0 && (val == AF_INET || val == AF_INET6)) {
```

Comment:
You'd ask, why not check with `socket_->IsUDS()`... Well its always false 😆 `UringSocket::Accept` just doesn't propagate this flag in the file-descriptor meta bits

You'd ask, why not just propagate this flag... Well, we `DCHECK(fd & UDS == 0)` in some places, like `RemoteEndpoint()` which is called for all connections 

So we have two bugs, but they cancel out each other. Not sure if I should fix both or just keep this fix

Reply:
Yes... I'll fix it in this pr, so lets keep it as a draft for now